stdlib/LinearAlgebra/src/uniformscaling.jl

```diff
@@ -480,6 +480,9 @@ Array(s::UniformScaling, dims::Dims{2}) = Matrix(s, dims)
 Diagonal{T}(s::UniformScaling, m::Integer) where {T} = Diagonal{T}(fill(T(s.λ), m))
 Diagonal(s::UniformScaling, m::Integer) = Diagonal{eltype(s)}(s, m)
 
+dot(A::AbstractMatrix, J::UniformScaling) = tr(A)*J.λ
+dot(J::UniformScaling, A::AbstractMatrix) = J.λ*tr(A)
```

Comment:
This seems safer, since the generic implementation of `dot` calls `dot` on the individual elements of the matrices. Also this makes sure that the first argument is conjugated correctly.
```suggestion
dot(A::AbstractMatrix, J::UniformScaling) = dot(tr(A), J.λ)
dot(J::UniformScaling, A::AbstractMatrix) = dot(J.λ, tr(A))
```